Mobile Buy SDK/Mobile Buy SDK/Models/BUYManagedObject.m

```diff
@@ -91,6 +92,83 @@ - (NSDictionary *)jsonDictionaryForCheckout
 	return self.JSONDictionary;
 }
 
+#pragma mark - Dynamic Property Accessor Resolution
+
+NS_INLINE NSString *KeyForSelector(SEL selector, BOOL *isSetter)
+{
+	NSString *name = NSStringFromSelector(selector);
+	BOOL setter = [name hasPrefix:@"set"];
+	if (isSetter) {
+		*isSetter = setter;
+	}
+	return setter ? [[name substringWithRange:NSMakeRange(3, name.length - 4)] lowercaseString] : name;
+}
+
+- (BOOL)respondsToSelector:(SEL)selector
+{
+	return [super respondsToSelector:selector] || [self isPropertyAccessor:selector];
+}
+
+static NSMutableDictionary *signatureCache;
```

Comment:
I did not profile it, so it is premature optimization. The speed improvement from the cache is probably marginal for generating the signatures. It is also used to return quickly from `-isPropertyAccessor:`.